Molio/Source/Presentation/MyInfo/ViewModel/MyInfoViewModel.swift

```diff
@@ -20,4 +41,61 @@ final class MyInfoViewModel: ObservableObject {
     var isPossibleConfirmButton: Bool {
         isPossibleNickName && isPossibleDescription
     }
+    
+    func didSelectImage(_ item: PhotosPickerItem) {
+        item.loadTransferable(type: Data.self) { result in
+            DispatchQueue.main.async { [weak self] in
+                guard case .success(let data) = result else { return }
+                guard let data = data else { return }
+                self?.userSelectedImageData = data
+                self?.profileImageUpdateType = .update(data)
+            }
+        }
+    }
+    
+    func removeProfileImage() {
+        userImageURL = nil
+        userSelectedImageData = nil
+        profileImageUpdateType = .remove
+    }
+    
+    func updateUser() async throws {
+        guard let userID = try currentUserIdUseCase.execute() else { return }
+        
+        await MainActor.run {
+            self.isLoading = true
+        }
+        
+        do {
+            try await userUseCase.updateUser(
+                userID: userID,
+                newName: userNickName,
+                newDescription: userDescription,
+                imageUpdate: profileImageUpdateType
+            )
+            await MainActor.run {
+                self.alertState = .successUpdate
+                self.showAlert = true
+                self.isLoading = false
+            }
+        } catch {
+            self.alertState = .failureUpdate
+            self.showAlert = true
+            self.isLoading = false
+        }
+    }
```

Comment:
P3;
혹시 이렇게 작성할 수도 있나요?

```suggestion
    @MainActor
    func updateUser() async throws {
        guard let userID = try currentUserIdUseCase.execute() else { return }
        self.isLoading = true
        
        do {
            try await userUseCase.updateUser(
                userID: userID,
                newName: userNickName,
                newDescription: userDescription,
                imageUpdate: profileImageUpdateType
            )
            self.alertState = .successUpdate
            self.showAlert = true
            self.isLoading = false
        } catch {
            self.alertState = .failureUpdate
            self.showAlert = true
            self.isLoading = false
        }
    }
```

Reply:
넵 똑같이 작동할 것 같습니다 !!